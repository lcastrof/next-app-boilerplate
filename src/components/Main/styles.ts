import styled from 'styled-components';

export const Wrapper = styled.main`
  background-color: ${({ theme }) => theme.colors.background};
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding-inline: 3em;
`;

export const Title = styled.h1`
  font-size: 5rem;
  color: ${({ theme }) => theme.colors.text};
`;

export const Subtitle = styled.h2`
  font-size: 3rem;
  background: -webkit-linear-gradient(#8062ff, #6c63ff);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
`;

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(60rem, 100%);
`;
