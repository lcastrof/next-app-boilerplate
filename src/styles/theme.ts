export const theme = {
  colors: {
    primary: '#8062ff',
    secondary: '#6c63ff',
    background: '#121212',
    text: '#dedeff'
  }
};

export type AppTheme = typeof theme;
