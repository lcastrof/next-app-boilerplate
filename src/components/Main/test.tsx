import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { Main } from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Main />
      </ThemeProvider>
    );

    expect(
      screen.getByRole('heading', { name: /next boilerplate/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the colors correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Main />
      </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('background-color', '#121212');
  });
});
