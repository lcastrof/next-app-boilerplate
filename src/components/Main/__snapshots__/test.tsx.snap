// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #121212;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-inline: 3em;
}

.c1 {
  font-size: 5rem;
  color: #dedeff;
}

.c2 {
  font-size: 3rem;
  background: -webkit-linear-gradient(#8062ff,#6c63ff);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c3 {
  margin-top: 3rem;
  width: min(60rem,100%);
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Next boilerplate
  </h1>
  <h2
    class="c2"
  >
    Nice hacking!
  </h2>
  <img
    alt="Man standing next to rocket watching it launch."
    class="c3"
    src="img/launch.svg"
  />
</main>
`;
